### PR TITLE
RUN-5262 Uncaught js exception in main process

### PR DESCRIPTION
**Pull request by licui3936**
This PR is to fix uncaught js exception in main process found by GS.  

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [ ] manual tests are [changed or added](https://github.com/openfin/test_apps)
- [ ] relevant documentation is [changed or added](https://github.com/hadoukenio/js-adapter)
- [x] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [x] PR release notes describe the change in a way relevant to app-developers
- [ ] Link to new tests added
- [x] PR has assigned reviewers

### CODE REVIEW

File: src/browser/api/system.js [hunk 1]
```diff
@@ -163,15 +163,14 @@ export const System = {
         };
 
         electronApp.vlog(1, `clearCache ${JSON.stringify(storages)}`);
-        clearCacheInvoked(true);
 
-        defaultSession.clearCache(() => {
-            defaultSession.clearStorageData(cacheOptions, () => {
-                resolve();
+        grantAccess(() => {
+            defaultSession.clearCache().then(() => {
```
Comment (MichaelMCoates) [inline src/browser/api/system.js]: Async/await this please!

Comment (licui3936) [reply src/browser/api/system.js]: Good catch! just updated it.

File: src/browser/api/system.js [hunk 2]
```diff
@@ -163,15 +163,14 @@ export const System = {
         };
 
         electronApp.vlog(1, `clearCache ${JSON.stringify(storages)}`);
-        clearCacheInvoked(true);
 
-        defaultSession.clearCache(() => {
-            defaultSession.clearStorageData(cacheOptions, () => {
-                resolve();
+        grantAccess(async () => {
```
Comment (datamadic) [inline src/browser/api/system.js]: Do we need at `catch` or a `finally` on the `clearCache` call or will that throw and resolve (nack)? Also, do we need that await there?

Comment (licui3936) [reply src/browser/api/system.js]: I will add try-catch block.  
await is needed here.

Comment (datamadic) [reply src/browser/api/system.js]: Perfect, lgtm. Once the tests clear I'll merge

**Pull request merged**